billing-service/src/main/java/org/egov/demand/service/BillServicev2.java

```diff
@@ -284,26 +288,48 @@ public BillResponseV2 generateBill(GenerateBillCriteria billCriteria, RequestInf
 				.tenantId(billCriteria.getTenantId())
 				.email(billCriteria.getEmail())
 				.consumerCode(consumerCodes)
-				.isPaymentCompleted(false)
```

Comment:
This change is old. Search only demands with unpaid dues rather than fetching all and filtering. Please explain if there is a need for this.

Reply:
Yes, because for fully paid properties this API throwing the exception and that message is displaying in the UI as 
"No Demands Found for the given bill generate criteria"

Reply:
An empty response can be returned for all scenarios. no need for error in any case.

Reply:
@kavi-egov Here when there are no unpaid demands are the fetchbill API throwing an error and hence we added this criterion to fetch demand for a bill, what are other scenarios, we also do not know as of now. We can think of other scenarios later, as of now, this work I feel. If something else is somewhere to be done for this fix. Please guide Rafeek to get this done.